gym/spaces/sequence.py

```diff
@@ -50,29 +54,49 @@ def is_np_flattenable(self):
         return False
 
     def sample(
-        self, mask: Optional[Tuple[Optional[np.ndarray], Any]] = None
+        self,
+        mask: Optional[Tuple[Optional[Union[np.ndarray, int]], Optional[Any]]] = None,
     ) -> Tuple[Any]:
         """Generates a single random sample from this space.
 
         Args:
             mask: An optional mask for (optionally) the length of the sequence and (optionally) the values in the sequence.
                 If you specify `mask`, it is expected to be a tuple of the form `(length_mask, sample_mask)` where `length_mask`
-                is either `None` if you do not want to specify any restrictions on the length of the sampled sequence (then, the
-                length will be randomly drawn from a geometric distribution), or a `np.ndarray` of integers, in which case the length of
-                the sampled sequence is randomly drawn from this array. The second element of the tuple, `sample` mask
-                specifies a mask that is applied when sampling elements from the base space.
+                is
+                - `None` The length will be randomly drawn from a geometric distribution)
+                - `np.ndarray` of integers, in which case the length of the sampled sequence is randomly drawn from this array.
+                - `int` for a fixed length sample
+                The second element of the mask tuple `sample` mask specifies a mask that is applied when
+                sampling elements from the base space. The mask is applied for each feature space sample.
 
         Returns:
             A tuple of random length with random samples of elements from the :attr:`feature_space`.
         """
         if mask is not None:
             length_mask, feature_mask = mask
         else:
-            length_mask = None
-            feature_mask = None
+            length_mask, feature_mask = None, None
+
         if length_mask is not None:
-            length = self.np_random.choice(length_mask)
+            if np.issubdtype(type(length_mask), np.integer):
+                assert (
+                    0 <= length_mask
+                ), f"Expects the length mask to be greater than zero, actual value: {length_mask}"
```

Comment:
I guess the text should say greater than or equal to zero if we're allowing zero length sequences.